src/h3lib/include/iterators.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file iterators.h
+ * @brief Iterator structs and functions for the children of a cell,
+ * or cells at a given resolution.
+ */
+
+#ifndef ITERATORS_H
+#define ITERATORS_H
+
+#include <stdint.h>
+
+#include "h3api.h"
+
+/*
+Iter_Child: struct for iterating through the descendants of
+a given cell
+ */
+typedef struct {
+    H3Index h;
+    int parentRes;  // parent resolution
+    int fnz;        // first nonzero digit (this digit skips `1` for pentagons)
```

Comment:
I think these names may be a little too short for use in a structure used in multiple places. I would replace `h` with `currentIndex` or similar. I would replace `fnz` with `_firstNonZeroDigit` and `parentRes` with `_parentRes` (underscore to indicate they are internal to the iterator implementation)

Reply:
I agree with `_parentRes` and what I'm currently calling `_skipDigit`, but I think `h` is fairly idiomatic within the library, no?